src/CheckBox/index.js

```diff
@@ -17,41 +18,59 @@ import { Label, HiddenCheckbox, StyledCheckbox } from './elements';
 class CheckBox extends PureComponent {
   /** Internal state. */
   state = {
-    checked: this.props.checked, // eslint-disable-line
+    isChecked: this.props.isDefaultChecked, // eslint-disable-line
   };
 
   /**
-   * Component Did Update
-   * Method invoked immediately after updating occur.
+   * Returns true if requested prop exists
+   *
+   * @param {string} prop - requested property name
+   *
+   * @return {boolean}
    */
-  componentDidUpdate(prevProps) {
-    const { checked } = this.props;
-
-    // Update state only if props have changed
-    if (prevProps.checked !== checked) {
-      this.setState({
-        checked,
-      });
-    }
-  }
+  isControlled = prop => {
+    const props = this.props;
+
+    return props.hasOwnProperty(prop);
+  };
+
+  /**
+   * Returns the value a property from props if prop exists and from state otherwise
+   *
+   * @param {string} key - requested property name
+   *
+   * @return {*} value of the property
+   */
+  getControllableValue = key => {
+    const props = this.props;
+    const state = this.state;
+
+    return this.isControlled(key) ? props[key] : state[key];
+  };
 
   /**
    * Handle Change
    * fired when checkbox is triggered and state changes.
    */
   handleChange = () => {
-    const { onChange, disabled } = this.props;
+    const { onChange, isDisabled } = this.props;
 
-    if (disabled) {
+    if (isDisabled) {
       return;
     }
 
-    this.setState(
-      prevState => ({ checked: !prevState.checked }),
-      () => {
-        onChange;
-      },
-    );
+    const isChecked = this.getControllableValue('isChecked');
+
+    if (this.isControlled('isChecked')) {
+      onChange && onChange(isChecked);
+    } else {
+      this.setState(
+        prevState => ({ isChecked: !prevState.isChecked }),
+        () => {
+          onChange && onChange(isChecked);
```

Comment:
If component is uncontrolled, you may want to provide the new value of `isChecked` to the parent. Here, you send back the old one.